src/App.tsx

```diff
@@ -15,48 +16,117 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  const isAlphabeticSort = sortType === SortType.ALPHABET;
+  const isLengthSort = sortType === SortType.LENGTH;
+
+  if (isAlphabeticSort) {
+    visibleGoods.sort((g1, g2) => g1.localeCompare(g2));
+  }
+
+  if (isLengthSort) {
+    visibleGoods.sort((g1, g2) => g1.length - g2.length);
+  }
```

Comment:
I would probably use `switch-case` for `sortType`, but its up to you.

Reply:
agree, let's apply switch-case, it's right the case to use it, put it inside compare callback